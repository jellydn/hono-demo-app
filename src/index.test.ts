import app from ".";

describe("Test the application", () => {
  it("Should return 200 response", async () => {
    const res = await app.request("http://localhost/");
    expect(res.status).toBe(200);
    const body = await res.text();
    expect(body).toMatchSnapshot();
  });

  it("Should return 404 response", async () => {
    const res = await app.request("http://localhost/notfound");
    expect(res.status).toBe(404);
    const body: { message: string } = await res.json();
    expect(body).toMatchSnapshot();
    expect(body.message).toBe("Not Found");
  });
});
