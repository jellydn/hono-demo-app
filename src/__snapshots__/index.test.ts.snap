// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the application Should return 200 response 1`] = `"<h1>Hello Hono!</h1>"`;

exports[`Test the application Should return 404 response 1`] = `
Object {
  "message": "Not Found",
  "ok": false,
}
`;
